openchrom/plugins/net.openchrom.xxd.process.supplier.templates/src/net/openchrom/xxd/process/supplier/templates/peaks/PeakTransfer.java

```diff
@@ -333,7 +323,40 @@ private void transfer(IPeak peakSource, IChromatogram<? extends IPeak> chromatog
 		}
 	}
 
-	private double calculateSigma(IChromatogramCSD chromatogramCSD, IPeak peak) {
+	private IPeak fitPeak(IChromatogramCSD chromatogramCSD, Point maxPosition, double percentageIntensity, int startRetentionTime, int stopRetentionTime) {
+
+		int centerRetentionTime = (int)maxPosition.getX();
+		float intensity = (float)(maxPosition.getY() * percentageIntensity);
+		int centerScan = chromatogramCSD.getScanNumber(centerRetentionTime);
+		double sigma = calculateSigma(chromatogramCSD);
+		Gaussian gaussian = new Gaussian(intensity, centerScan, sigma);
+		IPeak peakSink = createDefaultGaussPeakNormal(chromatogramCSD, startRetentionTime, stopRetentionTime, gaussian, intensity);
+		if(peakSink == null) {
+			return null;
+		}
+		while(!isWithinBounds(peakSink, chromatogramCSD)) {
+			sigma = sigma / 2;
+			gaussian = new Gaussian(intensity, centerScan, sigma);
+			peakSink = createDefaultGaussPeakNormal(chromatogramCSD, startRetentionTime, stopRetentionTime, gaussian, intensity);
+			if(peakSink == null) {
+				return null;
+			}
+		}
+		return peakSink;
+	}
+
+	private boolean isWithinBounds(IPeak peakSink, IChromatogramCSD chromatogramCSD) {
+
+		IPeakModel peakModel = peakSink.getPeakModel();
+		for(int rt : peakModel.getRetentionTimes()) {
```

Comment:
using rt is fine, but I prefer to use the full word, here: `for(int retentionTime : peakModel.getRetentionTimes()) {`